### PR TITLE
[Kubernetes Plugin] Update the Kubernetes plugin documentation

### LINKED ISSUE #14014: [K8s plugin] Improve the Kubernetes plugin documentation
The current [Kubernetes plugin documentation](https://backstage.io/docs/features/kubernetes/overview) has some info missing:

1. Didn't mention `objectTypes` config which controls the default resource types to fetch, it only exists in source code and changelog which is not easy to find.
2. [The RBAC](https://backstage.io/docs/features/kubernetes/configuration#role-based-access-control) should be updated to reflect the requirement for metrics. Since the `skipMetricsLookup` is false by default, then access to the `pods.metrics.k8s.io` resource is required, but it's not mentioned anywhere in the doc. Besides, the error message is not particularly helpful, it could lead to some confusion (the PR https://github.com/backstage/backstage/pull/13966 tries to address this issue).
3. The documentation is a bit hard to read in general. e.g. It's hard to notice that `catalog`, `localKubectlProxy` and `config` are on the same level, and below are the subfields of `config`.

image

I would love to open a PR to fix 1 and 2, but I'm not sure what could be done for the general readability of the documentation. I suppose some restructuring is needed but I don't want to jump the gun, therefore I opened this issue for discussion.

### PR DESCRIPTION
**Pull request by namco1992**
- Add the `objectTypes` config
- Add the access needed to fetch pod metrics in the RBAC chapter

Signed-off-by: Mengnan Gong <namco1992@gmail.com>

## Hey, I just made a Pull Request!

Closes #14014.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/features/kubernetes/configuration.md
Comment (djamaile) [inline docs/features/kubernetes/configuration.md:412]: You need to add these terms to `.github/vale/Vocab/Backstage/accept.txt` for the pipeline to pass 😄

File: docs/features/kubernetes/configuration.md
Comment (djamaile) [inline docs/features/kubernetes/configuration.md:479]: This comment is a bit out of place with the rest of the example.

**Review by djamaile — CHANGES_REQUESTED** [review]
Nice! Just few comments

File: docs/features/kubernetes/configuration.md
Comment (djamaile) [reply docs/features/kubernetes/configuration.md:479]: Yes 😄

Comment (namco1992) [reply docs/features/kubernetes/configuration.md]: Updated, thank you!

File: docs/features/kubernetes/configuration.md
Comment (djamaile) [inline docs/features/kubernetes/configuration.md:485]: Shouldn't we add `-watch` here as well?

Comment (jamieklassen) [reply docs/features/kubernetes/configuration.md:485]: Actually `list` is the only verb Backstage ever uses. If you stare at this expression:

https://github.com/backstage/backstage/blob/b3fc6211936118dfeceeb3fe6ce767095b6342e5/plugins/kubernetes-backend/src/service/KubernetesFetcher.ts#L184-L190

It always ends in a plural

**Review by djamaile — APPROVED** [review]

File: docs/features/kubernetes/configuration.md
Comment (namco1992) [reply docs/features/kubernetes/configuration.md:485]: @jamieklassen thanks for the clarification. I believe the `watch` is not really required, but I didn't touch the existing docs. I can make it consistent if it's more preferred.

**Comment by namco1992** [comment]
Hello 👋  is there anything pending on my side to get it merged?

**Comment by djamaile** [comment]
lets wait for another reviewer to shave a look as well 😄

File: docs/features/kubernetes/configuration.md
Comment (benjdlambert) [inline docs/features/kubernetes/configuration.md:437]:
```suggestion
    - configmaps
    - deployments
    - limitranges
    - pods
    - services
    - statefulsets
```

**Review by benjdlambert — APPROVED** [review]
Just a small comment nothing major, but approving to unblock the merging 👍

**Pull request merged**
